state/status.go

```diff
@@ -191,6 +191,17 @@ func probablyUpdateStatusHistory(st *State, globalKey string, doc statusDoc) {
 	}
 }
 
+func eraseStatusHistory(st *State, globalKey string) error {
+	history, closer := st.db().GetCollection(statusesHistoryC)
+	defer closer()
+	historyW := history.Writeable()
+
+	if _, err := historyW.RemoveAll(bson.D{{"globalkey", globalKey}}); err != nil {
```

Comment:
Yeah, that makes sense - essentially the same thinking that prompted me to move eraseHistory in here, but more so. The reflection code would at least have the advantage of failing noisily if the field changed, rather than the situation here where everything seemed fine except for all of the table scans. I'll add a constant for this anyway.